pkg/kubelet/volume/empty_dir/empty_dir.go

```diff
@@ -197,36 +201,51 @@ func (ed *emptyDir) GetPath() string {
 
 // TearDown simply discards everything in the directory.
 func (ed *emptyDir) TearDown() error {
-	switch ed.medium {
-	case api.StorageTypeDefault:
-		return ed.teardownDefault()
-	case api.StorageTypeMemory:
-		return ed.teardownTmpfs()
-	default:
-		return fmt.Errorf("unknown storage medium %q", ed.medium)
+	return ed.TearDownAt(ed.GetPath())
+}
+
+// TearDownAt simply discards everything in the directory.
+func (ed *emptyDir) TearDownAt(dir string) error {
+	// Figure out the medium.
+	if medium, err := ed.mediumer.GetMedium(dir); err != nil {
+		glog.Errorf("TIM: error from GetMedium: %v", err)
+		return err
+	} else {
+		switch medium {
+		case mediumMemory:
+			ed.medium = api.StorageTypeMemory
```

Comment:
@thockin This is where the tests are failing on Linux but passing on OS X.

"medium == 1" here because of this line:   https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/volume/empty_dir/empty_dir_linux.go#L35

SetUp goes through ed.setupDefault() while TearDown goes through ed.teardownTempfs()